react/react_forms/README.md

```diff
@@ -1,387 +1,477 @@
 
-Then to access the value, we either use `.value` for input & select fields, or `.checked` for checkbox fields. 
-
-> What happens if you use .value on a checkbox?
+*Discussion question: What would happen if you used .value to key into a checkbox field's value?*
 
 For example:
 
 ```js
-  const handleFormSubmit = (event) => {
-    event.preventDefault();
-    console.log(event.target.firstname) // <input type="text" placeholder="First Name" name="firstname">
-    console.log(event.target.firstname.value) // Jimmy
-    console.log(event.target.notARobot.checked) // true
+handleFormSubmit = (event) => {
+  event.preventDefault();
+  const { firstName, notARobot } = event.target
+  console.log(firstName) // This is the node itself. You should see something like: <input type="text" placeholder="First Name" name="firstName">
+  console.log(firstName.value) // Jimmy
+  console.log(notARobot.checked) // true
 ```
 
-We can use our favorite destructuring syntax to shorten this up.
+Our final version of this function should look something like this:
 
 ```js
 const handleFormSubmit = (event) => {
   event.preventDefault();
+
   // get all the name properties
-  let {firstname, lastname, title, notARobot} = event.target
+  const {firstName, lastName, title, notARobot} = event.target;
   
-  //validate them 
-  if(allFieldsValid(firstname.value, lastname.value, title.value, notARobot.checked)) {
-    alert("Form submitted successfully!")
-  }
-  else {
-    alert("Please fill out the form completely")
+  // validate them 
+  if (allFieldsValid(firstname.value, lastname.value, title.value, notARobot.checked)) {
+    alert("Form submitted successfully!");
+  } else {
+    alert("Please fill out the form completely");
   }
-}
+};
 ```
 
-We also need to change how `allFieldsValid()` works. Since we only have access to the fields in the scope of the function that handles `onSubmit`, we need to rewrite our function to accept arguments instead of referencing something outside of it.
+Because we're using the event, rather than our component's state, we also need to change how `allFieldsValid()` works. Since we only have access to the fields in the scope of the function that handles `onSubmit`, we need to rewrite `allFieldsValid` to accept arguments instead of referencing something outside of it.
 
 ```js
 const allFieldsValid = (...args) => {
   // loop through all arguments that were passed in
   // if any are falsy, return false
-  // otherwise return true
-  for(let i = 0; i < args.length; i++) {
-    if(!args[i]) {
-      return false
+  // otherwise, return true
+  args.forEach(arg => {
+    if (!arg) {
+      return false;
```

Comment:
I don't believe we can return out of a forEach

Reply:
I believe that's correct @coreyladovsky -- we will cut this section anyway but good catch.

Reply:
Y'all got me - I was just mindlessly converting to ES6, hah.